php/php.editor/test/unit/src/org/netbeans/modules/php/editor/completion/PHP84CodeCompletionTest.java

```diff
@@ -0,0 +1,484 @@
+    }
+
+    public void testNewWithoutParentheses_26b() throws Exception {
+        checkCompletion("        }->publicMet^hod7()::publicStaticTestMethod7();");
+    }
+
+    public void testNewWithoutParentheses_26c() throws Exception {
+        checkCompletion("        }->publicMethod7()::^publicStaticTestMethod7();");
+    }
+
+    public void testNewWithoutParentheses_26d() throws Exception {
+        checkCompletion("        }->publicMethod7()::publicStaticTes^tMethod7();");
+    }
+
+    public void testNewWithoutParenthesesTyping01() throws Exception {
+        checkCompletion("new Test()::^");
+    }
+
+    public void testNewWithoutParenthesesTyping02() throws Exception {
+        checkCompletion("new Test()::public^");
+    }
+
+    public void testNewWithoutParenthesesTyping03() throws Exception {
+        checkCompletion("new Test()->^");
+    }
+
+    public void testNewWithoutParenthesesTyping04() throws Exception {
+        checkCompletion("new Test()->publicTe^");
+    }
+
+    public void testNewWithoutParenthesesTyping05() throws Exception {
+        checkCompletion("        $example = new Example()->^");
+    }
+
+    public void testNewWithoutParenthesesTyping06() throws Exception {
+        checkCompletion("        $example = new Example()->priva^");
+    }
+
+    public void testNewWithoutParenthesesTyping07() throws Exception {
+        checkCompletion("        $example = new Example()::^");
+    }
+
+    public void testNewWithoutParenthesesTyping08() throws Exception {
+        checkCompletion("        $example = new Example()::public^");
+    }
+
+    public void testNewWithoutParenthesesTyping09() throws Exception {
+        checkCompletion("        $example = new Example()->test->^");
+    }
+
+    public void testNewWithoutParenthesesTyping10() throws Exception {
+        checkCompletion("        $example = new Example()->test->publicTe^");
+    }
+
+    public void testNewWithoutParenthesesTyping11() throws Exception {
+        checkCompletion("        $example = new Example()->test::^");
+    }
+
+    public void testNewWithoutParenthesesTyping12() throws Exception {
+        checkCompletion("        $example = new Example()->test::public^");
+    }
+
+    public void testNewWithoutParenthesesAnonTyping01() throws Exception {
+        checkCompletion("}::^;");
+    }
+
+    public void testNewWithoutParenthesesAnonTyping02() throws Exception {
+        checkCompletion("}->^;");
+    }
+
+    public void testNewWithoutParenthesesAnonTyping03() throws Exception {
+        checkCompletion("}?->^;");
+    }
+
+    // TODO: add tests for typing anonymous class without ";"
+    // e.g.
+    // echo new class() {
+    //     public const string CONSTANT = "constant";
+    // }::^
+    // in the above case, the statement is broken (it has a syntax error)
+    // so, the anonymous class is not parsed correctly i.e. we can't get members
+    // we have to sanitize an error part
```

Comment:
I didn't fix this because it's so hard/difficult to fix it.

Reply:
OK, no problem, I think.